src/utils/check_astyle.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/sh
+
+./src/utils/astyle.sh
+
+git diff
+
+git diff --quiet HEAD 2>/dev/null
+
+if [ $? -gt 0 ]; then
+     echo "Style error! Run astyle: ./src/utils/astyle.sh"
+
+     git reset --hard 2>/dev/null 1>&2
```

Comment:
Why do we need to reset and clean? Can't it lead to users losing their changes?

Reply:
Good point. This script is meant to check if the code complies to the style guide, show the infringing bits and throw an error if not, but leave no side effects.